vernac/pvernac.ml

```diff
@@ -39,17 +52,33 @@ module Vernac_ =
       ] in
       Pcoq.grammar_extend main_entry None (None, [None, None, rule])
 
-    let command_entry_ref = ref noedit_mode
+    let select_tactic_entry spec =
+      match spec with
+      | None -> noedit_mode
+      | Some ename -> find_tactic_entry ename
+
     let command_entry =
       Pcoq.Entry.of_parser "command_entry"
-        (fun strm -> Pcoq.Entry.parse_token_stream !command_entry_ref strm)
+        (fun strm -> Pcoq.Entry.parse_token_stream (select_tactic_entry !command_entry_ref) strm)
 
   end
 
+let set_tactic_entry oname =
+  command_entry_ref := oname
+
 let main_entry = Vernac_.main_entry
 
-let set_command_entry e = Vernac_.command_entry_ref := e
-let get_command_entry () = !Vernac_.command_entry_ref
+type frozen_t = string option
```

Comment:
I am not sure it is worth duplicating the state here twice. Indeed, `Vernacstate` contains this info so I think we should make the API here functional, something such as `parse_vernac_or_tactic : ?tmode:string -> ....` Clients can indeed be sure they properly thread the `Vernacstate.t`.

Reply:
> `Vernacstate` contains this info

Really? Can you point me to where the proof mode is stored in `Vernacstate`? Is it in the proof state?

Reply:
Indeed it should be in the proof state, right?

Reply:
Are you saying it is already in the proof state, or that we should put it there?

Reply:
I think all the needed information is already there.

Reply:
Sorry, I couldn't find the name of the proof mode there. Where is it, more precisely?

Reply:
In `Proof_global`:
```
type pstate = {
  terminator : proof_terminator CEphemeron.key;
  endline_tactic : Genarg.glob_generic_argument option;
  section_vars : Constr.named_context option;
  proof : Proof.t;
  mode : proof_mode CEphemeron.key;
  universe_decl: UState.universe_decl;
  strength : Decl_kinds.goal_kind;
}

type t = pstate list
```

Reply:
Of course this needs tweaking due to the Ephemeron, etc... but the info is already there.

Reply:
Oh! So this PR was actually removing the mode from the proof state before I took it over. That's why I couldn't see it.

Reply:
Yeah I guess where to place it is kinda open, but I think I do prefer that part of the parser to be functional and have the state living in upper layers.

Reply:
That way, clients of the parser are indeed forced to declare the provenance of whether they think we are parsing proofs or not.